### ISSUE TITLE
defining your own function breaks "opa eval .. -- profile", yet policy evaluates fine in VSCode

### ISSUE THREAD

**Issue by andriyfomenko** [open-policy-agent/opa#2334]
Reduced sample

```
rules = [ 1, 2, 3 ]

default authorize = false
authorize {
  rule := rules[_]
  allow(rule)
}

allow(rule) {
  true
}
```

Environments:
1. VSCode
2. CLI: "opa eval -i ../input -d test/generated --profile --format=pretty data.authorize"

# Expected Behavior

"authorize" should evaluate to "true" in any execution context, VSCode and CLI.

# Actual Behavior

1. VScode executes just fine, as expected.

2. CLI always returns "false", but if "allow" is commented out or converted to be NOT a function, evaluates to "true" as appropriate

# Additional Info

OPA ver 0.19.1

**Comment by patrick-east** [comment]
I tried to reproduce this but am not seeing the "false" result with the reduced sample: 

```
{12:45} ~ ❯ cat policy.rego
package test

rules = [ 1, 2, 3 ]

default authorize = false
authorize {
  rule := rules[_]
  allow(rule)
}

allow(rule) {
  true
}
{12:45} ~ ❯ opa eval -d policy.rego -f pretty --profile 'data.test.authorize'
true
+------------------------------+--------+
|            METRIC            | VALUE  |
+------------------------------+--------+
| timer_rego_load_files_ns     | 352513 |
| timer_rego_module_compile_ns | 335403 |
| timer_rego_module_parse_ns   | 252042 |
| timer_rego_query_compile_ns  | 63247  |
| timer_rego_query_eval_ns     | 333310 |
| timer_rego_query_parse_ns    | 7061   |
+------------------------------+--------+
+-----------+----------+----------+---------------------+
|   TIME    | NUM EVAL | NUM REDO |      LOCATION       |
+-----------+----------+----------+---------------------+
| 132.228µs | 3        | 3        | policy.rego:8       |
| 72.506µs  | 3        | 3        | policy.rego:12      |
| 61.468µs  | 1        | 1        | data.test.authorize |
| 44.527µs  | 1        | 1        | policy.rego:3       |
| 17.292µs  | 1        | 3        | policy.rego:7       |
+-----------+----------+----------+---------------------+
```

Can you share more detailed steps to reproduce the issue?

**Comment by andriyfomenko** [comment]
It looks like as I reduced sample to the gist of it, the issue could not be reproduced anymore...

Here is the ZIP of the full original example: https://drive.google.com/file/d/1aHo180RRUWSNGeoqrf9vU6lw3fufhKpw/view?usp=sharing

please view the authz.rego file

we are evaluating "authorize" rule, line 30

if you execute that in VSC, it will evaluate to "true" on the test data provided

if you try executing the CLI (opa eval ... -profile ..), it will evaluate to "false" on the very same input.json and other data

I found that commenting out lines 38/39 (function calls) will make for both VSC and CLI to match results

then I introduced the rule of 

ttt(poli_id) {
  true
}

and made a call of that rule in line 38 => CLI "broke" (evaluated to "false"), while VSC would keep evaluating to "true"

at this point, I concluded this looks suspiciously close to "bug", but as you pointed above, the reduced sample is not confirming it, so at this point I'm sharing the full example

sincerely,
A.F.

**Comment by patrick-east** [comment]
What version of the VSCode plugin do you have installed? Just want to make sure it matches up with my system.

Can you share the CLI commands used? I see
```
opa eval -i ../input -d test/generated --profile --format=pretty data.authorize
```
But i'm not sure what directory structure that's using (I don't see a `test/generated` path in the source)

With the source in vscode and on the CLI I'm getting the same results for both (using the equivelent CLI command as what the VSCode plugin is using for "Evaluate Selection")

VSCode:
![image](https://user-images.githubusercontent.com/3614287/80263074-52024500-8644-11ea-8715-4cfc74d80ee5.png)

Console:
```
{15:57} /t/2334:master ◈ ❯ opa eval -b . -i ./input.json -f pretty --profile 'data.authz.authorize'
true
+------------------------------+-----------+
|            METRIC            |   VALUE   |
+------------------------------+-----------+
| timer_rego_data_parse_ns     | 492695052 |
| timer_rego_load_bundles_ns   | 518570292 |
| timer_rego_module_compile_ns | 3749211   |
| timer_rego_module_parse_ns   | 513001    |
| timer_rego_query_compile_ns  | 42218     |
| timer_rego_query_eval_ns     | 1197497   |
| timer_rego_query_parse_ns    | 10917     |
+------------------------------+-----------+
+-----------+----------+----------+----------------------+
|   TIME    | NUM EVAL | NUM REDO |       LOCATION       |
+-----------+----------+----------+----------------------+
| 355.267µs | 8        | 5        | common.rego:13       |
| 134.017µs | 4        | 4        | authz.rego:67        |
| 110.539µs | 8        | 5        | authz.rego:73        |
| 95.72µs   | 2        | 2        | authz.rego:34        |
| 64.095µs  | 2        | 2        | authz.rego:22        |
| 50.333µs  | 4        | 4        | authz.rego:70        |
| 41.435µs  | 8        | 0        | common.rego:9        |
| 38.964µs  | 1        | 1        | data.authz.authorize |
| 34.623µs  | 2        | 2        | authz.rego:60        |
| 34.022µs  | 2        | 2        | authz.rego:25        |
+-----------+----------+----------+----------------------+
```

**Comment by andriyfomenko** [comment]
Patrick, it looks like I lost an ability to reproduce it in my environment.

I used to have two directories with sym-linked .rego files and it was reproducible when I opened the ticket, it was in this semi-working state what I complained above, but now this problem is gone.

On the positive side, now "opa eval .. " positively breaks when I opened it in the directory where the data is local and .rego files are symlinked.

I used that setup for faster test on a secondary set of data (the one I provided in ZIP), but since then I simly merged everything togetehr into one directory structure and it is not a problem for me any longer.

sorry for the trouble
